### ISSUE TITLE
KT-23413 IndexOutOfBoundsException on nested delegated properties and `provideDelegate`

### ISSUE THREAD

**Issue by bamboo** [gradle/kotlin-dsl#784]
https://youtrack.jetbrains.com/issue/KT-23413

**Comment by mkobit** [comment]
Ran into something like this as well with the following code in a project:

```kotlin
// some subprojectval generateCloudProperties = tasks.findByPath(":generateCloudProperties")!!
// #1
// val outputDestination: File by generateCloudProperties.extra
// val generateCloudPropertiesOutputFile = outputDestination

// #2
// val generateCloudPropertiesOutputFile = generateCloudProperties.extra["outputDestination"] as File

// #3
val generateCloudPropertiesOutputFile = run {
    val outputDestination: File by generateCloudProperties.extra
    outputDestination
}
```

1 and 2 both succeed, 3 fails

**Comment by eskatos** [comment]
Confirmed it is fixed in 1.2.60-eap-75, closing
